svm-decode/src/arrow_converter.rs

```diff
@@ -0,0 +1,581 @@
+use crate::deserialize::{DynType, DynValue};
+use anyhow::{anyhow, Context, Result};
+use arrow::array::{Array, ListArray};
+use arrow::{
+    array::{builder, ArrowPrimitiveType, RecordBatch, StructArray},
+    buffer::{NullBuffer, OffsetBuffer},
+    datatypes::*,
+};
+use std::sync::Arc;
+
+/// Converts a vector of dynamic values into an Arrow array based on the specified type.
+///
+/// # Arguments
+/// * `param_type` - The type information for the values to convert
+/// * `values` - Vector of optional dynamic values to convert
+///
+/// # Returns
+/// * `Result<Arc<dyn Array>>` - The converted Arrow array wrapped in an Arc
+pub fn to_arrow(param_type: &DynType, values: Vec<Option<DynValue>>) -> Result<Arc<dyn Array>> {
+    match param_type {
+        DynType::I8 => to_number::<Int8Type>(8, &values),
```

Comment:
actual_bits/expected_bits doesn't apply here because all types match. It is needed in ethereum because there can be arbitrary bit-width integers.

So can remove the bit-width checks